target/linux/lantiq/dts/FRITZ7312.dts

```diff
@@ -0,0 +1,140 @@
+/dts-v1/;
+
+#include "ar9.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "FRITZ7312 - 1&1 WLAN-MODEM";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+
+		led-internet = &info_green;
+		led-dsl = &power;
+		led-wifi = &wlan;
+	};
+
+	memory@0 {
+		reg = <0x0 0x4000000>;
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x1000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					ath9k_cal: partition@0 {
+						label = "urlader";
+						reg = <0x00000 0x20000>;
+						read-only;
+					};
+
+					partition@20000 {
+						label = "firmware";
+						reg = <0x20000 0xf60000>;
+					};
+
+					partition@f80000 {
+						label = "tffs (1)";
+						reg = <0xf80000 0x40000>;
+						read-only;
+					};
+
+					partition@fc0000 {
+						label = "tffs (2)";
+						reg = <0xfc0000 0x40000>;
+						read-only;
+					};
+				};
+			};
+		};
+
+		gpio: pinmux@E100B10 {
+			pinctrl-names = "default";
+			pinctrl-0 = <&state_default>;
+
+			state_default: pinmux {
+			};
+		};
+
+		etop@E180000 {
+			mtd-mac-address = <&ath9k_cal 0xa91>;
+			mtd-mac-address-increment = <(-2)>;
+		};
+
+		ifxhcd@E101000 {
+			status = "okay";
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <100>;
+
+		rfkill {
+			label = "rfkill";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+		dect {
+			label = "dect";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_PHONE>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+		power: power {
+			label = "fritz7320:green:power";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+		voice {
+			label = "fritz7320:green:fon";
+			gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
+		};
+		dect {
+			label = "fritz7320:green:dect";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+		};
+		wlan: wlan {
+			label = "fritz7320:green:wlan";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+		info_green: info_green {
```

Comment:
According to the manual, there should be an info red led as well. Would you please add this one as well.

Reply:
I'll test leds. On device board it didn't see a 2nd led @info - yay, manual says there is a red one.

Reply:
tried gpios 24 to 56 in /source/target/linux/lantiq/dts/FRITZ7312.dts using LuCi after compile, aka:

		info_red {
			label = "fritz7312:red:info";
			gpios = <&gpio 24 GPIO_ACTIVE_LOW>;
		};
		info_red2 {
			label = "fritz7312:red:info";
			gpios = <&gpio 25 GPIO_ACTIVE_LOW>;
		};
	...
		info_redX {
			label = "fritz7312:red:info";
			gpios = <&gpio 56 GPIO_ACTIVE_LOW>;
		};

no red led could be activated so far.

if -> gpio 34 GPIO_ACTIVE_LOW -> connection loss an Lan1 (has only one)
if -> gpio 24 GPIO_ACTIVE_LOW -> bootloop
if -> 35 GPIO_ACTIVE_HIGH -> led "broken"
if -> 54+55+56 GPIO_ACTIVE_HIGH -> led broken

Reply:
I neither wasn't able to find the GPIO of the mentioned red info led in the sources. But I found references to some of the mentioned GPIOs:

- GPIO#34: ext phy1 reset
- GPIO#54: dect reset

Seem to me the manual is wrong and there isn't a red info led.

Reply:
ill take "info_red" out for now